compilers/concrete-compiler/compiler/include/concretelang/Dialect/FHE/IR/FHEOps.td

```diff
@@ -523,6 +523,25 @@ def FHE_ReinterpretPrecisionEintOp: FHE_Op<"reinterpret_precision", [Pure, Unary
     let results = (outs FHE_AnyEncryptedInteger);
 }
 
+def FHE_ChangePartitionEintOp: FHE_Op<"change_partition", [Pure, UnaryEint]> {
+
+    let summary = "Change partition if necessary.";
+
+    let description = [{
+      Changing the partition of a ciphertext.
+      If necessary, it keyswitch the ciphertext to a different key having a different set of parameters than the original one.
+
+      Example:
+      ```mlir
+        %new_eint = "FHE.change_partition"(%eint): (!FHE.eint<16>) -> (!FHE.eint<16>)
+    ```
+    }];
+
+    let arguments = (ins FHE_AnyEncryptedInteger:$input);
+    let results = (outs FHE_AnyEncryptedInteger);
+    let hasVerifier = 1;
```

Comment:
For the partition attribute, can we have strings ? like the name of the partition.
With attribute name like src_partition, dst_partition.
With at least one of the two should be here.

Reply:
and who's supposed to set those names?

Reply:
Yes I'm ok with rudy, we need to known which are the source / target partitions else the optimization cannot known which part of the dag should be assigned to an already parametrized partition (the tfhe-rs one in our case). The frontend and/or the user could give this information.

Reply:
I'm wondering also how we should give the partition(s) parameters, any ideas?